### DISCUSSION TITLE
How to trigger a callback once data finished loading into the grid?

### DISCUSSION THREAD

**Discussion by scarletjack** [glideapps/glide-data-grid#261] [Q&A]
Title basically describes the question. Is there some way to implement an onLoaded() callback or something similiar?

I'd like to use it to set a default selected _and_ visible (already scrolled to) row. Unless there's a better way to do it? 

Doing something like 

```tsx
ref.current.scrollTo(0, selectRowId)
setSelected(...)
```

Works fine (edit: For selection), but doesn't scroll to the row coordinate

Thanks for any help!

**Comment by jassmith** [comment]
Ah hmmm let me figure that out. I suppose the ideal method would be to have the grid auto-scroll to the selected cell on load.

**Reply by scarletjack** [reply]
Not a very elegant solution, but for now I got around the issue by setting a default false boolean state during the start of `getCellContent` to true, using the useEffect hook on it to scroll to a specified index.

```tsx
  const getCellContent = useCallback(
    (cell) => {
      setFlag(true)
      ...
    }, [])
```

```tsx
  useEffect(() => {
    if (args.cellState === true) {
      ref.current.scrollTo(0, selectRowIdDeb)
    }
  }, [args.cellState])
  ```
  
The use case for this would be, as an example, when the data for the grid is updated via a popup form of some kind, and the grid reloads it's newest data for that row, the sorting might change for that specific row, and should be scrolled to and reselected.
